### PR TITLE
[RFC] go lang support as a build sytle

### PR DESCRIPTION
**Pull request by yopito**
golang support as a build style, and some packaging to illustrate its usage.
apart from various WIP stuff in packages, I'll be glad with some informal review of this build style.

### CODE REVIEW

**Comment by q66** [comment]
thanks, i don't think we need this right now but once things are a bit more stable we can work out some review

**Comment by q66** [comment]
i wasn't happy with the whole gcompat nonsense so i experimented with generating toolchains and at that point i was far enough to have a working template, so that's in the repo now, suggest dropping the toolchain from here and also drop gcompat because it makes no sense for chimera (any binaries you'd want to run with that would be linked with libgcc, and we don't have that and there is a good chance we never will)

we can try getting in the build style eventually (it looks like it needs a lot of work though)

the random software should be standalone pull requests

**Comment by yopito** [comment]
OK, i'll rework this PR, only with the go build style.

File: src/cbuild/util/golang.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        if ldflags:
+            myargs += ["-ldflags", f"{(' ').join(ldflags)}"]
+        if self.template.make_build_args:
+            myargs += self.template.make_build_args
+        myargs += args
+
+        return self._invoke("install", myargs, jobs, True, None, env, wrksrc)
+
+    def check(self):
+        myargs = ["-v"]
```
Comment (vazub) [inline src/cbuild/util/golang.py:102]: I would rather suggest to be silent by default, not verbose.

Comment (nekopsykose) [reply src/cbuild/util/golang.py:102]: iirc go test verbose doesn't even have anything useful except being loud- failures still print the failure and source line, so -v isn't useful, just spammy

**Comment by vazub** [comment]
I've tried the style and it works rather well overall. One thing that requires further investigation is that the app I've built did get debug symbols stripped, but a separate `-dbg` subpackage was not created.

File: src/cbuild/util/golang.py (hunk 1 above)
Comment (yopito) [reply src/cbuild/util/golang.py:102]: Thanks for your inputs. OK, I will remove it

File: src/cbuild/util/golang.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    return env
+
+class Golang:
+    def __init__(self, tmpl, jobs = None, env = {}, wrksrc = None):
+        self.template = tmpl
+        self.wrksrc = wrksrc
+        self.env = env
+        self.jobs = jobs
+        # -dbg packages fail to be generated
+        tmpl.options["debug"] = False
```
[the remark this answers is not in the document]
Comment (vazub) [reply src/cbuild/util/golang.py:24]: Removed the line and tested locally - works well, -dbg gets created. So just removing the line here helps.

**Review by vazub — CHANGES_REQUESTED** [review]

**Review by vazub — APPROVED** [review]
Should be fine to merge, unless @q66 has any additional concerns/requirements.

**Comment by q66** [comment]
i did a preliminary merge of this after some cleanup/adjustments, but without the documentation; only stable interfaces can be documented and i plan to not make this stable for a while (just like cargo)

**Pull request closed without merging**